core/src/main/scala/kafka/coordinator/transaction/TransactionCoordinator.scala

```diff
@@ -30,6 +30,13 @@ import org.apache.kafka.common.record.RecordBatch
 import org.apache.kafka.common.requests.TransactionResult
 import org.apache.kafka.common.utils.{LogContext, ProducerIdAndEpoch, Time}
 
+class TransactionStateTopicPartitionCountViaZooKeeper(zkClient: KafkaZkClient,
```

Comment:
Can we not simply have a method in the companion object that we can lift into a function?

Reply:
The logic to create a function `() => zkClient.getTopicPartitionCount(Topic.TRANSACTION_STATE_TOPIC_NAME).getOrElse(defaultValue)` is needed in 2 separate places (once in `TransactionCoordinator` and once in `TransactionStateManager`), so based on that I refactored it out into a common place here.  I can remove this class if you prefer.

Reply:
Yeah, I am suggesting you can still have it shared, but as a method in the companion object of one of the classes. And then you can have a simpler method name like `transactionStatePartitionCount` or something.

Reply:
To clarify, it's a bit of an anti pattern to have a whole class when a method would do in Scala.

Reply:
I left a comment in the other PR.  We do not want to get rid of startup, or have a callback function.  See #10008.  I agree with @ijuma that we should consolidate these into one PR to keep the discussion in one place.

Reply:
Closing this PR -- I consolidated these changes into https://github.com/apache/kafka/pull/10008 so we can review them together.